### ISSUE TITLE
OpenAsync seems to work synchronously

### ISSUE THREAD

**Issue by borisgeller** [NVentimiglia/WebSocket.Portable#2]
Steps to reproduce:
1) Disconnect internet (turn off wi-fi and mobile data, pull cable from the computer etc)
2) Call OpenAsync

Expected result: call returns quickly
Actual result: call never returns, program is stuck

**Comment by NVentimiglia** [comment]
Thanks, I will look at it later this week.

I should add I am working on a replacement project due to issues with TLS implementations on mono. The replacement project will use native Android/iOS sockets via a binding library.

**Comment by swanitzek** [comment]
Any updates regarding your replacement project? Will you make it available on GitHub?

**Comment by NVentimiglia** [comment]
Thanks for reminding me. I have something, and will publish something this weekend.

**Comment by NVentimiglia** [comment]
The updating is taking more time than anticipated. 

Little info on what I have to do. Basically due to the issues with mono sockets, Ive decided that going native is the best bet. So I will be needing to offload all websocket work to a native client and writing a api for accessing the websocket from C#. This is much like how modern http works.

My first attempt was to use the Xamarin binding libraries provided by square. These packages mostly work, but, do cause segfaults from time to time during disconnection. I reason these segfaults are caused by the 'magic' of the binding library itself.

My second attempt is to not use a magical binding library, but, handle the binding myself. This will allow me to use the looper and other android / ios threading mechanisms to guarantee thread health. This approach is what I use for my Unity3d projects, and works fine.

In short, I have an action plan, but, it is still a work in progress.

**Comment by swanitzek** [comment]
Nicholas thank you so much for your effort. I really appreciate it.

Keep up the good work.

**Comment by NVentimiglia** [comment]
Update, Blocked on a binding library issue.

http://forums.xamarin.com/discussion/59685/android-binding-library-package-does-not-exist-cannot-find-symbol?new=1

**Comment by NVentimiglia** [comment]
https://github.com/NVentimiglia/Websockets.Pcl

Please try this out. It has limited testing, but is a radically simpler app.
